File: 204-count-primes/count-primes.cpp
class Solution {
public:
    
    int countPrimes(int n) {
        vector<bool> p(n + 1, true);
        p[0] = false;
        p[1] = false;
        for (int i = 2; i * i <= n; i++)
        {
            if (p[i])
            {
                for (int j = i * i; j <= n; j += i)
                {
                    p[j] = false;
                }
            }
        }
        return (int)count(p.begin(), p.end() - 1, true);
    }
};

File: 204-count-primes/count-primes.java
class Solution {
    public int countPrimes(int n) {
        if (n < 1) return 0;

        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;

        for (int p = 2; p * p <= n; p++) {
            if (isPrime[p]) {
                for (int i = p * p; i <= n; i += p) {
                    isPrime[i] = false;
                } 
            }
        }

        int res = 0;
        for (int i = 0; i < n; i++) {
            if (isPrime[i]) res++;
        }

        return res;
    }
}